src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1935,27 +1935,38 @@ def check_process(processName):
     return False
 
 
-def get_custom_locations_oid(cmd):
+def get_custom_locations_oid(cmd, cl_oid):
     try:
         sp_graph_client = get_graph_client_service_principals(cmd.cli_ctx)
         sub_filters = []
         sub_filters.append("displayName eq '{}'".format("Custom Locations RP"))
         result = list(sp_graph_client.list(filter=(' and '.join(sub_filters))))
         if len(result) != 0:
-            return result[0].object_id
-        else:
-            logger.warning("Unable to fetch oid of 'custom-locations' app. Proceeding without enabling the feature.")
+            if cl_oid is not None and cl_oid != result[0].object_id:
+                logger.debug("The 'Custom-locations' OID passed is different from the actual OID({}) of the Custom Locations RP app. Proceeding with the correct one...".format(result[0].object_id))
+            return result[0].object_id  # Using the fetched OID
+
+        if cl_oid is None:
+            logger.warning("Failed to enable Custom Locations feature on the cluster. Unable to fetch Object ID of Azure AD application used by Azure Arc service. Try enabling the feature by passing the --custom-locations-oid parameter directly. Learn more at https://aka.ms/CustomLocationsObjectID")
```

Comment:
May I ask why this log is warning level?

Reply:
Because, we don't want to error out at this stage.

This is a feature enablement part which is switched on by default, and if not able to be enabled, we print a warning to the user.

Reply:
Got it, thanks~